### ISSUE TITLE
sync not working for new items

### ISSUE THREAD

**Issue by warpr** [lastpass/lastpass-cli#166]
I am having trouble syncing changes, so far what I've been able to replicate consistently is:

1. lpass login kuno@followupboss.com
2. lpass sync  # Any items which are in the upload-queue are synced correctly at this point.
3. lpass add some site
4. lpass sync # This second sync will fail

Using ```LPASS_LOG_LEVEL=7``` with current master (efb90bf98e10bac1b419050637e6491c74f099cb) these runs look like:

```
<7> [1459866553.288939] UQ: starting queue run
<7> [1459866553.289526] UQ: processing job upload-queue/14598662590000
<7> [1459866553.289544] UQ: posting to loglogin.php
<7> [1459866553.891790] UQ: result 0 (http_code=200)
<7> [1459866553.891832] UQ: succeeded
<7> [1459866553.892228] UQ: processing job upload-queue/14598663940000
<7> [1459866553.892247] UQ: posting to show_website.php
<7> [1459866554.500875] UQ: result 0 (http_code=200)
<7> [1459866554.500953] UQ: succeeded
<7> [1459866554.501414] UQ: processing job upload-queue/14598664560000
<7> [1459866554.501441] UQ: posting to show_website.php
<7> [1459866555.035888] UQ: result 0 (http_code=200)
<7> [1459866555.035978] UQ: succeeded
<7> [1459866555.625699] UQ: queue run complete
```
Second run:

```
<7> [1459866626.250535] UQ: starting queue run
<7> [1459866626.251015] UQ: processing job upload-queue/14598666260000
<7> [1459866626.251031] UQ: posting to show_website.php
<7> [1459866634.877561] UQ: result 35 (http_code=0)
<7> [1459866634.877607] UQ: attempt 2, sleeping 1 seconds
<7> [1459866635.877746] UQ: posting to show_website.php
<7> [1459866641.492930] UQ: result 35 (http_code=0)
<7> [1459866641.492967] UQ: attempt 3, sleeping 8 seconds
<7> [1459866649.493056] UQ: posting to show_website.php
<7> [1459866649.606544] UQ: result 35 (http_code=0)
<7> [1459866649.606591] UQ: attempt 4, sleeping 64 seconds
```

**Comment by bcopeland** [comment]
Thank you for the report.  Hmm, that is odd: result 35 means the connection is being dropped at TLS connection establishment time, so it's not even getting to the web servers.  Moreover I'm not seeing any errors on the server logs for your IP.  I assume these requests did fail after 5 or so attempts?  If you want to retry the transactions, you can try copying them back from the failed directory to the upload queue directory in ~/.lpass and resync.

If it were something like an upstream proxy, or your clocks being wrong, we should expect that the first set of transactions also do not work.  So I'm a bit stumped.

**Comment by dnorthup-ums** [comment]
I have been able to reproduce this behavior, running 82c6594d0.

**Comment by bcopeland** [comment]
@dnorthup-ums your log also shows result 35?

**Comment by dnorthup-ums** [comment]
@bcopeland Yep. I even double-checked the upstream's certs via QUALYS SSL Labs, and as I suspected they look fine to me.

**Comment by bcopeland** [comment]
@dnorthup-ums can you verify that you have access to https://lastpass.com from the same IP at the same time?  Just want to rule out automated blocking from our firewall.

**Comment by dnorthup-ums** [comment]
Ok, I have verified that I'm not getting auto-blocked by the firewall. It seems that virtually all initial queue runs fail, yet virtually all "second queue runs" (after moving back from the "upload-fail" directory to the "upload-queue" one) succeed.
Thinking it might be a random race issue, I negative offset the timestamps used for naming the queue files. That didn't make any difference.
I also tried disabling the agent, thinking there might be a bug hiding in the session data. That didn't make any difference either.
Next, I'm thinking about testing what some have referred (in other circles) to the "nuclear" option: disable TLS certificate verification. I suspect it won't fix it, but it may give better insight into what is actually going wrong. I'm also probably going to enable some of the libcurl debugging output.

**Comment by dnorthup-ums** [comment]
Ok, so if I specify "--sync=no" and then manually sync I can edit an existing one and sync the first shot. Now to try it with a new item.

**Comment by dnorthup-ums** [comment]
Adding with "sync=no" and then running a separate sync job works, every time I've tried it.

**Comment by bcopeland** [comment]
@dnorthup-ums thanks for the hint...

I was finally able to see this on someone else's RHEL6 machine (still doesn't happen on my Debian box).  It looks like there is a weird interaction here with NSS and the fact that we fork for the upload-queue for the subsequent edits.  The first update goes through in the parent, initializing some NSS stuff, the subsequent updates are queued for later in the child.  Then libcurl has issues validating certificates in the child process.  So workarounds might include reinitializing libcurl in the child, baking the cert back in, or something else.  Still working through it at the moment.

**Comment by bcopeland** [comment]
OK, I have reproduced in a CentOS VM and will push a fix shortly.
